source/Octopus.Cli/Util/RepositoryExtensions.cs

```diff
@@ -152,6 +154,68 @@ public static Task<RunbookResource> FindByNameOrIdOrFail(this IRunbookRepository
                 nameOrId,
                 $" in {project.Name}");
         }
+        
+        internal static async Task<ChannelResource> FindGitRefChannelByNameOrIdOrFail(this IOctopusAsyncRepository repo,
+            ProjectResource project,
+            string nameOrId,
+            string gitRef)
+        {
+
+            var resourceTypeIdPrefix = "Channels";
+            var resourceTypeDisplayName = "channel";
+            var enclosingContextDescription = $" in {project.Name}";
+            var skipLog = false;
+
+            ChannelResource resourceById;
+            if (!Regex.IsMatch(nameOrId,
+                $@"^{Regex.Escape(resourceTypeIdPrefix)}-\d+$",
+                RegexOptions.Compiled | RegexOptions.IgnoreCase))
+                resourceById = null;
+            else
+                try
+                {
+                    
+                    resourceById = await repo.Channels.Beta().Get(project, nameOrId, gitRef);;
+                }
+                catch (OctopusResourceNotFoundException)
+                {
+                    resourceById = null;
+                }
+
+            ChannelResource resourceByName;
+            try
+            {
+                // Duplicates missing FindByName in Client for Beta Repository
+                resourceByName = await repo.Projects.Beta()
+                    .GetAllChannels(project, gitRef)
+                    .ContinueWith(d => 
+                        d.Result.SingleOrDefault(named =>
+                            string.Equals((named.Name ?? string.Empty).Trim(), nameOrId, StringComparison.OrdinalIgnoreCase))
+                        ,TaskContinuationOptions.NotOnFaulted)
+                    .ConfigureAwait(false);
+
+            }
+            catch (OctopusResourceNotFoundException)
+            {
+                resourceByName = null;
```

Comment:
is this worth logging in these catches as if we throw exception further down it might be harder to tell why from that stack trace, or would it just be noise in the logs?